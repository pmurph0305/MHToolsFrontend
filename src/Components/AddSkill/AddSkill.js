import React, { useState } from "react";
import PropTypes from "prop-types";

import InputCheckbox from "../InputCheckbox/InputCheckbox";
import InputField from "../InputField/InputField";
import InputText from "../InputText/InputText";
import InputTextArea from "../InputTextArea/InputTextArea";

import "./AddSkill.scss";

/**
 * AddSkill - Returns a component that contains inputs and text areas for creating a new coping skill.
 * @callback onAddSkill Callback when add skill button is clicked
 */
export const AddSkill = ({ onAddSkill }) => {
  const [addNewSkill, setNewSkill] = useState("");

  const addSkill = () => {
    onAddSkill();
    setNewSkill("");
  };

  return (
    <div className="add-skill-container">
      {!addNewSkill && (
        <InputField
          placeholder={"Enter a title for a new coping skill"}
          onClick={element => setNewSkill(element.value)}
          buttonTitle={"Create New Coping Skill"}
        />
      )}
      {addNewSkill && (
        <>
          <InputText
            defaultValue={addNewSkill}
            inputLabel="Coping Skill Title"
            placeholder="Enter a title for a new coping skill."
            idAndName="add_skill_title"
          />
          <InputTextArea
            idAndName="add_skill_description"
            inputLabel="Coping Skill Description"
            placeholder="Enter a description on how the coping skill is used."
          />
          <div className="add-skill-extras-container">
            <InputCheckbox idAndName="add_skill_share" label="Share Skill" />
            <div className="add-skill-buttons-container">
              <button className="add-skill-button" onClick={addSkill}>
                Add skill
              </button>
              <button
                className="cancel-skill-button"
                onClick={() => setNewSkill("")}
              >
                Cancel
              </button>
            </div>
          </div>
        </>
      )}
    </div>
  );
};

AddSkill.propTypes = {
  onAddSkill: PropTypes.func.isRequired
};

export default AddSkill;
