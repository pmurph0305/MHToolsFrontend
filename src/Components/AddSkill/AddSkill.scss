@import "../../styles/shared.scss";

.add-skill-button {
  @extend %ButtonBase;
  background-color: $success-background-color;
  color: $primary-color;
}

.cancel-skill-button {
  @extend %ButtonBase;
  background-color: $error-background-color;
  color: $primary-color;
  margin-left: 1rem;
}

.add-skill-extras-container {
  margin: 1rem;
  display: flex;
  justify-content: space-between;
}
