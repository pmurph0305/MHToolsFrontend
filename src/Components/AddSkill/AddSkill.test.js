import { mount } from "enzyme";
import React from "react";
import AddSkill from "./AddSkill";

describe("AddSkill tests", () => {
  let mockOnAddSkill = jest.fn(val => val);
  const wrapper = mount(<AddSkill onAddSkill={mockOnAddSkill} />);

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
