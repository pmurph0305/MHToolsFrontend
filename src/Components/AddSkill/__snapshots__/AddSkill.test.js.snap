// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSkill tests matches snapshot 1`] = `ReactWrapper {}`;
